src/silvimetric/resources/storage.py

```diff
@@ -226,31 +229,53 @@ def open(self, mode:str='r', timestamp=None) -> tiledb.SparseArray:
         finally:
             tdb.close()
 
-    def get_history(self):
-        af = tiledb.array_fragments(self.config.tdb_dir)
-        with self.open('r') as r:
-            meta = dict(r.meta)
+    def get_history(self, start_time: np.datetime64, end_time: np.datetime64,
+                    bounds: Bounds, name:str=None):
 
+        from ..resources import ShatterConfig
+
+        af = tiledb.array_fragments(self.config.tdb_dir, True)
+        m = [ ]
         for idx in range(len(af)):
 
-            begin = af[idx].timestamp_range[0]
-            if len(af) > idx + 1:
-                end = af[idx+1].timestamp_range[1]
+            # if processes are too short, begin and end times can be the same
+            # so it's necessary to adjust time bounds so we can grab information
+
+            if idx == 0:
+                begin = 0
             else:
-                end = af[idx].timestamp_range[1]
+                begin = af[idx-1].timestamp_range[1]
+
+            end = af[idx].timestamp_range[1]
 
             with self.open('r', (begin, end)) as r:
-                for key in r.meta.keys():
-                    if meta[key] is not None:
-                        if not isinstance(meta[key], list):
-                            meta[key] = [ meta[key] ]
-                        if not isinstance(r.meta[key], list):
-                            meta[key] = meta[key] + [ r.meta[key] ]
-                        else:
-                            meta[key] = meta[key] + r.meta[key]
+                try:
+                    if not bool(r.meta['shatter']):
```

Comment:
what if `'shatter' not in r.meta`? We should check and protect here. A db that hasn't had shatter operations is going to error.

Reply:
That's what this `try, except` is testing for, we're looking for if `KeyError` is thrown.